common/priorityQueue.py

```diff
@@ -0,0 +1,81 @@
+import bisect
+
+class PriorityQueue:
+    def __init__(self):
+        self._queue = []
```

Comment:
Not sure if this would be useful, but have you considered using a heapq here instead of a standard list? https://docs.python.org/2/library/heapq.html